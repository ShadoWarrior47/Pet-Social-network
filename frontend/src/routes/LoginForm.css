/* General container styling */
.Login-container div {
    max-width: 600px !important;
    margin: 0px auto;
    padding: 10px;
    font-family: Arial, sans-serif;
    min-width: 500px;
}

.Login-container form {
    display: flex;
    flex-direction: column;
    gap: 10px;
    padding: 10px;
    background-color: #f9f9f9;
    border: 1px solid #ddd;
    border-radius: 5px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.Login-container h1 {
    text-align: center;
    margin-bottom: 20px;
    font-size: 24px;
    color: #333;
}

.Login-container input {
    width: 100%;
    padding: 10px;
    font-size: 16px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.1);
    outline: none;
    transition: border-color 0.3s;
}

.Login-container input:focus {
    border-color: #007bff;
}

.Login-container label {
    font-size: 14px;
    font-weight: bold;
    margin-bottom: 5px;
    color: #555;
}

.Login-container button {
    padding: 10px 20px;
    font-size: 16px;
    color: white;
    background-color: #007bff;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.3s, transform 0.2s;
    width: 150px;
    margin: 0px auto;
}

.Login-container button:hover {
    background-color: #0056b3;
}

.Login-container button:active {
    transform: scale(0.98);
}