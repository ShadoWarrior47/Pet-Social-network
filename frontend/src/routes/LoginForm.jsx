import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./LoginForm.css";

function LoginForm({ login }) {
  const navigate = useNavigate();
  const INITIAL_STATE = {
    username: '',
    password: ''
  }

  const [formData, setFormData] = useState(INITIAL_STATE);
  const [isLoading, setIsLoading] = useState(false);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setIsLoading(true);
    const success = await login(formData);
    setIsLoading(false);
    if (success) {
      navigate("/");
    } else {
      alert("Login failed. Please check your credentials.");
    }
  };

  return (
    <div className="Login-container">
      <h1>Log In</h1>
      <form onSubmit={handleSubmit}>

        <div>
          <label htmlFor="username">
            Username :
          </label>
          <input
            id="username"
            name="username"
            type="text"
            placeholder="Username"
            autoComplete="current-username"
            value={formData.username}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="password">
            Password :
          </label>
          <input
            id="password"
            name="password"
            type="password"
            placeholder="Password"
            autoComplete="current-password"
            value={formData.password}
            onChange={handleChange}
            required
          />
        </div>

        <button type="submit" disabled={isLoading}>
          {isLoading ? "Logging in..." : "Log in"}
        </button>
      </form>
    </div>

  );
}

export default LoginForm;