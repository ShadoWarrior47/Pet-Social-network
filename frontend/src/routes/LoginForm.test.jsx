import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "./LoginForm";
import { vi } from "vitest";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <loginForm login={() => { }} />
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LoginForm login={() => { }} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("runs the login function on form submit", async function () {
    const loginMock = vi.fn();
    const { getByLabelText, getByText } = render(
        <MemoryRouter>
            <LoginForm login={loginMock} />
        </MemoryRouter>
    );

    const usernameInput = getByLabelText("Username :");
    const passwordInput = getByLabelText("Password :");
    const submitButton = getByText("Log in");

    fireEvent.change(usernameInput, { target: { value: "testuser" } });
    fireEvent.change(passwordInput, { target: { value: "password123" } });

    fireEvent.click(submitButton);

    expect(loginMock).toHaveBeenCalledWith({
        username: "testuser",
        password: "password123",
    });
});
