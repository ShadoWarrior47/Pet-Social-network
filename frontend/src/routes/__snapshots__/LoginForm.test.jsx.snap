// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login-container"
  >
    <h1>
      Log In
    </h1>
    <form>
      <div>
        <label
          for="username"
        >
          Username :
        </label>
        <input
          autocomplete="current-username"
          id="username"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password :
        </label>
        <input
          autocomplete="current-password"
          id="password"
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Log in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
